docs/project/breaking-change-process.md

```diff
@@ -0,0 +1,44 @@
+# Breaking change process
+
+Breaking changes make it difficult for people to adopt new versions of .NET. We should avoid making them. In the rare case that you believe a breaking change is unavoidable, you should follow the process outlined in this document. The intent of the process is to get feedback. This might result in previously unseen  opportunities to achieve your goals without breaks, clever ways to limit breaks to more niche scenarios, and (most importantly) better understand the scope of impact to .NET users.
+
+This process does not result in approval or denial of a breaking change request. It results in feedback, which might be varied or uniform. In the best case, the feedback is a call to action to you, that provides a direction on how you should proceed. The people that are responsible for reviewing your PR are expected to use the information collected by this process as part of their determination of whether your change is appropriate to merge.
+
+You should start this process as soon as possible, preferably before any code is written, in the design phase. The breaking change issue should be paired with a design document that helps reviewers understand the broader scope, goals and constraints of your change. If you don't yet have a design document, you are encouraged to first write such a document, to enable people giving feedback to develop a deeper level of understanding on your approach. If you do not do this, you might stall the process.
+
+There are people on the team that can help you work through a breaking change, if you want to engage with them before starting this more formal process. They are part of the [dotnet/compat team](https://github.com/orgs/dotnet/teams/compat) . Feel free to @mention them on issues to start a dialogue.
+
+## Process
+
+1. Create an issue that describes the breaking change
+   * Mark the issue with the [breaking-change](https://github.com/dotnet/runtime/labels/breaking-change) label
+   * Goals and motivation for the change
+   * Pre-change behavior
+   * Post-change behavior
+   * Errors or other behavior you can expect when running old code that breaks
+   * Workarounds and mitigations, including [AppContext switches](https://docs.microsoft.com/dotnet/api/system.appcontext)
```

Comment:
Do we have guidance somewhere indicating when a change justifies adding a switch? My impression was that our bar was high for this, to avoid reducing maintainability. 

Currently I see a couple for globalization invariant mode, a couple for threadpool, 3 for HTTP/2, a couple others. So it seems we use them for 1-2% of breaking changes.

Also we can presumably say that the switch would always be to switch off the default behavior rather than vice versa.

Reply:
I am happy to remove this. I saw somewhat similar text in doc Priya gave me, and was wanting to pull some of thinking from it. I look to you for guidance on this. I'm fully neutral on it.

Reply:
I think instead of removing this I think it would be much better if your doc could start talking about what criteria we use to decide whether we do an app compat switch or not.

Reply:
>  talk about what criteria we use

Here is one: https://github.com/dotnet/runtime/issues/8655#issuecomment-415103199